### PR TITLE
1812 soft-delete a course

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Deleting courses

### Guidance to review
```ruby
$ Provider.last.courses.count
26

$ Provider.last.courses.last.discard
-- deleting --

$ Provider.last.courses.count
25
```

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Test coverage

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Course - Delete a course (MCBE)](https://trello.com/c/YyJxBHzh/1813-delete-a-course)

File: app/models/provider.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@ class Provider < ApplicationRecord
           -> { published.latest_published_at },
           class_name: "ProviderEnrichment",
           inverse_of: 'provider'
-  has_many :courses
+  has_many :courses, -> { kept }
```
Comment (timabell) [inline app/models/provider.rb]: after discussion - we'll need to clear this scope for apiv1

Comment (fofr) [reply app/models/provider.rb]: Do we send UCAS courses that haven't been published yet? If we do – why?

**Comment by timabell** [comment]
moved to branch https://github.com/DFE-Digital/manage-courses-backend/tree/1813-delete-a-course

**Pull request closed without merging**